### PR TITLE
Check sku when creating linux ASP

### PR DESCRIPTION
**Pull request by LukaszStem**
Ensures sku is either basic or standard when creating a Linux app service plan.

### CODE REVIEW

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -419,13 +419,22 @@ def list_app_service_plans(resource_group_name=None):
     return plans
 
 
+def _linux_sku_check(sku):
+    tier = _get_sku_name(sku)
+    if tier == 'BASIC' or tier == 'STANDARD':
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: suggest `tier in ['BASIC', 'STANDARD']`

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py [hunk 2]
```diff
@@ -419,13 +419,22 @@ def list_app_service_plans(resource_group_name=None):
...
+        return
+    format_string = '{0} is not a valid sku for linux, please use one of the following: {1}'
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: Suggest `usage error: {0} is not a valid sku for linux plan, please use one of the following....`

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py [hunk 3]
```diff
@@ -419,13 +419,22 @@ def list_app_service_plans(resource_group_name=None):
...
+
+
 def create_app_service_plan(resource_group_name, name, is_linux, sku='B1', number_of_workers=None,
                             location=None):
     client = web_client_factory()
     sku = _normalize_sku(sku)
     if location is None:
         location = _get_location_from_resource_group(resource_group_name)
-
+    if is_linux is not None:
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: simplify to `if is_linux`, and please move his validation to the top

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py (hunk 1 above)
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: on a second thought, you should do  `!= 'FREE'`, this way if new sku comes up, you don't need to change the code

Comment (LukaszStem) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: what about all the other cases such as Premium (P1, P2, P3), Dynamic(D1, D), etc..

**Comment by tjprescott** [comment]
What message does the service return in this case? If it quickly returns that the SKU is invalid I would favor not hardcoding client-side validation.

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py (hunk 1 above)
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: My assumption is they should work?

**Comment by yugangw-msft** [comment]
@tjprescott, the error is really really bad per feedback from content team
`If your subscription is not in the FREE plan and you are using the –is-linux flag, it is ignored and the AppService is created on Windows/IIS  with no messaging to the developer. It may or may not fail btw. I see this as an issue or false positive. I f I ask for Linux, I should get it if allowed or be given a failure message.`

**Comment by troydai** [comment]
Shouldn't SDK provide the list of the supported SKUs instead of hard coding it in CLI? The vm services and sql services both provide API for fetching options list like vm images option, SQL service objectives options, and resource group location options.

**Comment by yugangw-msft** [comment]
@troydai, agreed, but keep in mind `webapp` service is really different in lots of aspects

**Comment by troydai** [comment]
If hardcoding the options is not avoidable, let's make it easier to manage. Please add a `const.py` module and list all the constant values including these options in it alongside with detailed comments. This will be helpful when the future maintainer needs to update the list.

**Comment by yugangw-msft** [comment]
> Please add a const.py module and list all the constant values including these options 
It is a a short term workaround before service gives out a good error, hence I don't think we need a file for it.

**Comment by tjprescott** [comment]
If we are introducing a workaround for a service deficiency, we should create an issue to track that so the workaround is removed once the service is fixed.

**Comment by yugangw-msft** [comment]
Issue #2672 is logged.

**Pull request merged**